src/QMCWaveFunctions/lcao/CuspCorrectionConstruction.cpp

```diff
@@ -211,55 +223,84 @@ void generateCuspInfo(int orbital_set_size,
   int start_mo = offset[Comm.rank()];
   int end_mo   = offset[Comm.rank() + 1];
   app_log() << "  Number of molecular orbitals to compute correction on this rank: " << end_mo - start_mo << std::endl;
-  for (int mo_idx = start_mo; mo_idx < end_mo; mo_idx++)
+
+#pragma omp parallel
   {
-    app_log() << "   Working on MO: " << mo_idx << std::endl;
+    ParticleSet localTargetPtcl(targetPtcl);
+    ParticleSet localSourcePtcl(sourcePtcl);
+
+    LCAOrbitalSet local_phi(phi);
+    local_phi.myBasisSet = phi.myBasisSet->makeClone();
+    local_phi.IsCloned   = true;
+    local_phi.C          = nullptr;
+    local_phi.setIdentity(false);
+
+    LCAOrbitalSet local_eta(eta);
+    local_eta.myBasisSet = eta.myBasisSet->makeClone();
+    local_eta.IsCloned   = true;
+    local_eta.C          = nullptr;
+    local_eta.setIdentity(false);
+
+// Specify dynamic scheduling explicitly for load balancing.   Each iteration should take enough
+// time that scheduling overhead is not an issue.
+#pragma omp for schedule(dynamic) collapse(2)
     for (int center_idx = 0; center_idx < num_centers; center_idx++)
```

Comment:
This should be the beginning of an actual lambda or function that controls what's available at thread scope. The signature and documentation should clearly show what is const/immutable at the thread scope and what is the shared object being updated. All the construction above would also cleary be in inside the thread "body" i.e. the lambda or function.